Also print if on wayland or Xorg in gather-system-info.sh

Lythenas opened the pull request:
This will also print a line:

```
Display server: Wayland
```

or

```
Display server: Xorg
```

depending on what the user is using.

---

Also regarding gnome 3.38 and the `gnome-extensions` command not being available: I found two fallbacks we could use:

1. `ls ~/.local/share/gnome-shell/extensions/` (lists also extensions that might not be enabled)
2. `gsettings get org.gnome.shell enabled-extensions` (list also system extensions)

I think I would prefer option 2. And maybe we would also want to print all extensions anyway and not just user extensions (as distro specific extensions are currently not visible).

Lythenas committed:
Also print if on wayland or Xorg

jtaala reviewed (approved):
Excellent!  tested in both X11 and Wayland.

jtaala commented:
> I think I would prefer option 2. And maybe we would also want to print all extensions anyway and not just user extensions (as distro specific extensions are currently not visible).

I agree.  Are you thinking to only use `gsettings` for a gnome 3.38 branch or switch to using `gsettings` regardless (would prefer to have consistency if possible).

Up to you, implement what you see fit and I'll test again and merge when you're ready.

Lythenas commented:
The problem with the `gsettings` approach I've noticed is that is can include uninstalled extensions too. E.g. for me this prints

```
$ gsettings get org.gnome.shell enabled-extensions
['nohotcorner@azuri.free.fr', 'appindicatorsupport@rgcjonas.gmail.com', 'TaskBar@zpydr', 'topicons-redux@pop-planet.info', 'pop-shell@system76.com', 'switcher@landau.fi', 'gnome-shell-mode@hedning:matrix.org', 'paperwm@paperwm-redux.github.com', 'paperwm@hedning:matrix.org']
```

But the only extensions I have actually installed and enabled are appindicator, switcher and paperwm@hedning:matrix.org (also no other builtin extensions). And some of the extensions aren't even installed anymore.

I will try to find a way to only print actually enabled extensions.

jtaala commented:
> I will try to find a way to only print actually enabled extensions.

Give it a go - if it's too finnicky, it's fine to leave the current one using `gnome-extensions` (since it's working well!) - for the `gnome-3.38` branch, could use the `ls` method?

Lythenas committed:
Add fallback for gnome-extensions command

Lythenas commented:
The implementation is a bit hacky, so please test it. It seems to work well on gnome 43. But I haven't tested it on older gnome versions.

@fumbles if you don't mind, could you test this updated script on gnome 3.32?

jtaala reviewed (approved):
Works well.  Good job.

jtaala commented:
@Lythenas, do you want me to give [@fumbles](https://github.com/fumbles) a couple of days to test in 3.32 - or should merge this PR into `develop` now (which I'm happy to do).

Lythenas commented:
Maybe wait a couple of days. But you can merge it when you want to do a release.

fumbles commented:
Sorry @Lythenas  @jtaala  I had forgotten about this :sweat:  Everything has been working with no issues. About to upgrade to fedora 38 soonish since it's approved so we'll see how that goes.

From: [Ubuntu forum](https://askubuntu.com/questions/1133782/command-to-list-installed-and-enabled-gnome-extensions)

Your locally installed gnome Shell extensions, i.e., these that are installed for your user only, can be listed with the command:

`ls ~/.local/share/gnome-shell/extensions/`

You can find out which extensions are enables by querying a dconf setting:
`gsettings get org.gnome.shell enabled-extensions`

System wide installed gnome-shell extensions are listed with the command
`ls /usr/share/gnome-shell/extensions/`


`gsettings get org.gnome.shell enabled-extensions | sed 's/,/\n/g'`
```
['background-logo@fedorahosted.org'
 'places-menu@gnome-shell-extensions.gcampax.github.com'
 'apps-menu@gnome-shell-extensions.gcampax.github.com'
 'appindicatorsupport@rgcjonas.gmail.com'
 'caffeine@patapon.info'
 'drive-menu@gnome-shell-extensions.gcampax.github.com'
 'dash-to-dock@micxgx.gmail.com'
 'clipboard-indicator@tudmotu.com'
 'utcclock@injcristianrojas.github.com'
 'paperwm@hedning:matrix.org']
```

Lythenas committed:
Simplify extension installed check

Lythenas commented:
Yes that is more or less what I'm doing. I'm listing all enabled extensions and then checking if they are actually installed. This was necessary on my machine because I guess I "uninstalled" some extension wrong (i.e. deleted folders). E.g. when switching from the paperwm-redux fork back to paperwm.

@jtaala I simplified the script a bit. But I think you can merge it after reviewing.

jtaala reviewed (approved):
Tested and works well.  Good work!

smichel17 commented on gather-system-info.sh:
```diff
@@ -20,6 +20,13 @@ fi
 gnome-shell --version
 
+echo -n "Display server: "
+if [ "${XDG_SESSION_TYPE}" = "wayland" ]; then
```
Just want to share one of my preferences for shell scripting, not asking for a change:
```suggestion
if test "${XDG_SESSION_TYPE}" = "wayland"; then
```
I really like the educational benefit of using `test` rather than single brackets.

- Helps build an intuition for "`if` just switches on the status code of the command" so it doesn't become a separate concept for people to understand when you want to use a non-builtin there.
- Possible to `man test` vs having to do a web search for how brackets work in [ba]sh
- Things like `-gt` being flags actually makes sense.
- Not significantly more verbose.

Lythenas replied:
Actually it is better to use `[[` instead of `[`. Because `[` is a synonym for `test` and therefore a command. And it is therefore also evaluated like a command, which has in some cases subtle, inconsistent or surprising behavior. `[[` is a keyword and therefore everything in `[[ ... ]]` is treated as "one thing" and evaluated in a more consistent way. Usually there is no difference at all. But in cases there is, it is usually not at all obvious what is going on.

Also using `[[` you can also use `<`, `>`, `=~` (for regex), `==` and `!=`. So in that sense it behaves more like other languages.

For most things bash I usually just refer to https://devhints.io/bash Since I can't remember everything and I think the syntax is not very intuitive. But you can also use `man bash` and search for `CONDITIONAL EXPRESSIONS`.
